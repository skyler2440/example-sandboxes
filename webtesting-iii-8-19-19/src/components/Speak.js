import React from "react";

const Speak = props => {
  return (
    <>
      <button data-testid="speak" onClick={props.speak}>
        Speak
      </button>
      <div>{props.message}</div>
    </>
  );
};

export default Speak;
