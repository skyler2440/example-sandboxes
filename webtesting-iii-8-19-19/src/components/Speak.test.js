import React from "react";
import renderer from "react-test-renderer"; // 1: install this npm module as a dev dependency
import { render, fireEvent } from "@testing-library/react";

import Speak from "./Speak";

describe("<Speak />", () => {
  // 2. write this test
  it("matches snapshot", () => {
    const tree = renderer.create(<Speak />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });

  // Implementation details ->
  //   - what props.speak is doing...
  //   - if props.speak is invoked...

  // What we _want_ to test:
  //    - When our user clicks that button... what happens in the app?
  it("displays a message when 'speak' button is clicked", () => {
    const speak = jest.fn();
    let messageMock = "";

    const { getByText } = render(<Speak speak={speak} message={messageMock} />);

    fireEvent.click(getByText(/speak/i));

    expect(speak).toHaveBeenCalled();
  });
});
