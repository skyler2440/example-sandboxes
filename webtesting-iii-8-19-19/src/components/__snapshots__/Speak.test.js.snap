// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Speak /> matches snapshot 1`] = `
Array [
  <button
    data-testid="speak"
  >
    Speak
  </button>,
  <div />,
]
`;
